core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -860,12 +829,12 @@ oppia.directive('conversationSkin', [
                     if (feedbackHtml) {
                       var stateHistory =
                         PlayerTranscriptService.getStateHistory();
-                      if (stateHistory.indexOf(newStateName) !== -1) {
+                      if (
+                        stateHistory.indexOf(nextCard.getStateName()) !== -1) {
```

Comment:
Define a function in PlayerTranscriptService itself: PlayerTranscriptService.hasEncounteredStateBefore(nextCard.getStateName());

I think something like this justifies the state name being on the card object, so I retract an earlier comment I made about that.

Reply:
Ok, will do this.

Reply:
As per initial comments, this won't be required, right?

Reply:
Yep.